lib/synapse/service_watcher/zookeeper_poll/zookeeper_poll.rb

```diff
@@ -52,7 +52,7 @@ def stop
       zk_teardown do
         # Signal to the thread that it should exit, and then wait for it to
         # exit.
-        @should_exit = true
+        @should_exit.set(true) unless @should_exit.nil?
```

Comment:
why only checking the nil here?  Seems not necessary as no one could call `stop` until `initialize` returns

Reply:
Good point! I added this in an earlier version where `@should_exit` was set in `start` instead of in `stop`.